### ISSUE TITLE
aria-label or test-id for for elements with no role ?

### ISSUE THREAD

**Issue by mo-alaa** [testing-library/dom-testing-library#1151]
Hi, I have a hidden input inside a button to upload files. 

```html
<button>
  <p>upload</p>
  <input hidden type='file'/>
</button>
```

 since the input 'file' doesn't have a role by default, should I add an `"aria-label" ` or a 
 `"data-testid"` for the input element?

Because as said [here](https://github.com/testing-library/dom-testing-library/issues/567#issuecomment-967789345) : 'The first rule of aria is don't use aria'.

Also testing library says this about using `data-testid `:  'Using data-testid attributes do not resemble how your software is used and should be avoided if possible'

so what should I pick ?

**Comment by eps1lon** [comment]
How is the hidden input used? If it's in a form whose data you use, test submitting it and check for data.

It's generally hard to give good advise without knowing the full UI.

**Comment by alexkrolick** [comment]
> since the input 'file' doesn't have a role by default

the default role of a file input is `button`

<img width="326" alt="image" src="https://user-images.githubusercontent.com/1571667/180823873-85097e2d-c159-4056-82ae-e5ef5ee7c7b8.png">

**Comment by MatanBobi** [comment]
@alexkrolick that's probably a chrome decision and not a spec definition so I guess it might not work in JSDOM.
Based on the [spec](https://www.w3.org/TR/2021/REC-html-aria-20211209/#docconformance), as @mo-alaa said, input type file has no implicit role:
![image](https://user-images.githubusercontent.com/12711091/181171776-c9123df4-2376-49cd-b0d3-d14b200fdf26.png)

**Comment by eps1lon** [comment]
Closing in favor of https://testing-library.com/discord. The documentation still applies. But concrete advise depends on the actual UI.
